api/v1alpha1/constants.go

```diff
@@ -3,14 +3,32 @@ package v1alpha1
 const (
 	ResourceHashAnnotation = "k8ssandra.io/resource-hash"
 
-	// StargateLabel is the distinctive label for objects created by the Stargate controller.
-	StargateLabel = "k8ssandra.io/stargate"
+	NameLabel      = "app.kubernetes.io/name"
+	NameLabelValue = "k8ssandra-operator"
 
-	DefaultStargateVersion = "1.0.30"
+	InstanceLabel = "app.kubernetes.io/instance"
+	VersionLabel  = "app.kubernetes.io/version"
 
-	PartOfLabel = "app.kubernetes.io/part-of"
+	ComponentLabel               = "app.kubernetes.io/component"
+	ComponentLabelValueCassandra = "cassandra"
+	ComponentLabelValueStargate  = "stargate"
 
+	CreatedByLabel                                = "app.kubernetes.io/created-by"
+	CreatedByLabelValueK8ssandraClusterController = "k8ssandracluster-controller"
+	CreatedByLabelValueStargateController         = "stargate-controller"
+
+	PartOfLabel      = "app.kubernetes.io/part-of"
 	PartOfLabelValue = "k8ssandra"
 
-	K8ssandraClusterLabel = "io.k8ssandra/cluster"
+	K8ssandraClusterLabel = "k8ssandra.io/cluster"
```

Comment:
I noticed this label had an inverted prefix: `io.k8ssandra` instead of `k8ssandra.io`. I fixed it.

Reply:
Java package naming conventions surfacing from my subconscious 😄

Reply:
I think every object that the operator creates should have the `K8ssandraClusterLabel` label. @adutra wdyt?

Reply:
I agree and I think it is already the case for `CassandraDatacenter` and `Stargate` objects. For `Deployment` and `Service` objects created by the Stargate controller, I think we should not set the label on every object, but only when the corresponding `Stargate` object also has the label. I will give it a try. Wdyt?

Reply:
My thinking was that every object created by the operator _is_ part of the K8ssandraCluster by adding the cluster label a user or admin can easily and conveniently query with a simple label selector to get a full inventory of all objects created by the operator for a given K8ssandraCluster. This is something on which we can iterate on and further discuss. I don't want it to hold up the PR.